### PR TITLE
[docs][yba] Advanced restore procedure

### PR DESCRIPTION
**Pull request by ddhodge**
Add details to the advanced restore procedure
PLAT-3714
PLAT-10127

@netlify /preview/yugabyte-platform/back-up-restore-universes/restore-universe-data/

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+Currently, YugabyteDB Anywhere supports backing up and restoring entire YSQL databases only. However, by performing some manual steps, you can restore single tables to a YSQL database using the following tools:
+
+- [ysql_dump](../../../manage/backup-restore/export-import-data/) - exports data from a YSQL database into a SQL script file.
+- [ysqlsh](../../../admin/ysqlsh/) - YugabyteDB SQL shell for interacting with YugabyteDB using YSQL.
+
+Suppose you have a problem with a single table in a database called source_db, and you want to restore the table from a backup taken previously. You would perform the following steps:
+
+1. Using a backup of source_db, restore the backup to a new database (call it restored_db). This can be done on the same or a different universe.
+1. In restored_db, use ysql_dump to export an SQL script file of the table to be restored.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:23]:
```suggestion
1. In restored_db, use ysql_dump to export a SQL script file of the table to be restored.
```

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+1. In source_db, use ysqlsh to drop the table with the same name (that is, the table you want restored from the backup), if it exists. (You only need to do this if the table with the same name exists on the database.)
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:24]: What do you mean by if the table exists? I thought we were talking about restoring this "problematic" table. So it should exist right?

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+The following example assumes the following basic setup:
+
+- a YSQL database named source_db.
+- source_db has 3 tables (table_1, table_2, and table_3) and the tables do not have any foreign key references.
+- source_db has been backed up using YugabyteDB Anywhere.
+
+### Restore the backup to a new database
+
+You can restore the backup to the same or to a different universe. This example assumes that you restored to the same universe.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:39]:
```suggestion
You can restore the backup to the same or to a different universe. This example assumes that you restored to the _same_ universe.
```

**Review by aishwarya24 — APPROVED** [review]
LGTM , just one question that I needed clarification.

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md (hunk 2 above)
Comment (ddhodge) [reply docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:24]: The problem might be the table was accidentally dropped.

File: docs/content/preview/manage/backup-restore/export-import-data.md [hunk 4]
```diff
@@ -40,16 +45,32 @@ To export a single database with all its tables, indexes, and other local artifa
...
 ./postgres/bin/ysql_dump -d mydb > backup/mydb-dump.sql
 ```
 
-For more details and a list of all available options, see [ysql_dump reference](../../../admin/ysql-dump/).
+## Export a single table
+
+To export a single database table, use the `ysql_dump` utility by executing the following command:
+
+```sh
+./postgres/bin/ysql_dump -t <table-name> <db-name> > <file>
```
Comment (vpatibandla-yb) [inline docs/content/preview/manage/backup-restore/export-import-data.md:62]: Why dont we use ./postgres/bin/ysql_dump -t <table-name> <db-name> -f <file> as specified in the docs: https://docs.yugabyte.com/preview/admin/ysql-dump/#f-file-file-file

File: docs/content/preview/manage/backup-restore/export-import-data.md [hunk 5]
```diff
@@ -40,16 +45,32 @@ To export a single database with all its tables, indexes, and other local artifa
...
+```
+
+- *table-name* is the name of the table to be exported.
+- *db-name* is the name of the database to be exported.
+- *file* is the path to the resulting SQL script file.
+
+For example, to export the `mytable` table of the `mydb` database into a file called `mytable-mydb-dump.sql` located in the `backup` folder, the command would be as follows:
+
+```sh
+./postgres/bin/ysql_dump -t mytable mydb > backup/mytable-mydb-dump.sql
```
Comment (vpatibandla-yb) [inline docs/content/preview/manage/backup-restore/export-import-data.md:72]: Make a change here based on the above change

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+    - *host* is the host IP of the universe node.
+    - *table-name* is the name of the table (table_1) to export.
+    - *database-name* is the name of the database (restored_db).
+    - *file-path* is the path to the resulting SQL script file.
+
+    For example:
+
+    ```sh
+    [yugabyte@ip-172-151-36-174 ~]$ ./ysql_dump -h 172.151.36.174 -t table_1 restored_db > /home/yugabyte/restored_db_table_1.sql
```
Comment (vpatibandla-yb) [inline docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:71]: Do not expose our internal IPs. Please remove it.

File: docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+### Drop the table from the source
+
+Before you can restore the table, you need to drop table_1 from the source_db database. To do this, you can use ysqlsh, located in the `/bin` directory of the YugabyteDB home directory.
+
+Do the following to drop a table from the source database source_db:
+
+1. Connect to one of the nodes of the universe with the source database (that is, the universe with source_db) and start ysqlsh. For example:
+
+    ```sh
+    [yugabyte@ip-172-151-36-174 ~]$ ./tserver/bin/ysqlsh -h 172.151.36.174
```
Comment (vpatibandla-yb) [inline docs/content/preview/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:83]: Same as above. Remove the internal ip details

File: docs/content/stable/manage/backup-restore/export-import-data.md [hunk 8]
```diff
@@ -40,16 +45,32 @@ To export a single database with all its tables, indexes, and other local artifa
...
 ./postgres/bin/ysql_dump -d mydb > backup/mydb-dump.sql
 ```
 
-For more details and a list of all available options, see [ysql_dump reference](../../../admin/ysql-dump/).
+## Export a single table
+
+To export a single database table, use the `ysql_dump` utility by executing the following command:
+
+```sh
+./postgres/bin/ysql_dump -t <table-name> <db-name> > <file>
```
Comment (vpatibandla-yb) [inline docs/content/stable/manage/backup-restore/export-import-data.md:62]: We can use ysql_dump -f file_name command

File: docs/content/stable/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+1. Connect to one of the nodes of the universe where you restored the backup (that is, the universe with restored_db) and change to the directory where ysql_dump is located. For example:
+
+    ```sh
+    [yugabyte@ip-172-151-36-174 ~]$ cd /home/yugabyte/tserver/postgres/bin
+    ```
+
+1. Use ysql_dump to export the table to a SQL script file. The syntax for dumping a single table is as follows:
+
+    ```sh
+    [yugabyte@ip-172-151-36-174 ~]$ ./ysql_dump -h <host> -t <table-name> <database-name> > <file-path>
```
Comment (vpatibandla-yb) [inline docs/content/stable/yugabyte-platform/back-up-restore-universes/restore-ysql-single-table.md:58]: Same as the comments given in preview

**Review by vpatibandla-yb — COMMENTED** [review]
Please address the comments shared.

File: docs/content/preview/manage/backup-restore/export-import-data.md (hunk 4 above)
Comment (ddhodge) [reply docs/content/preview/manage/backup-restore/export-import-data.md:62]: Not sure. Docs and examples taken from pg_dump docs https://www.postgresql.org/docs/11/app-pgdump.html

**Pull request merged**